streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableImpl.java

```diff
@@ -390,17 +390,22 @@ public String queryableStoreName() {
     @Override
     public KTable<K, V> suppress(final Suppressed<? super K> suppressed) {
         final String name;
+        // We need to burn an index even if name provided
```

Comment:
The suppression operator should not do this (evidenced by the fact that this change breaks the SuppressTopologyTest).

The KIP specifies the behavior when the name is present:
```
 * Use the specified name for the suppression node in the topology.
 * <p>
 * This can be used to insert a suppression without changing the rest of the topology names
 * (and therefore not requiring an application reset).
```